trin-core/src/types/validation.rs

```diff
@@ -0,0 +1,101 @@
+use anyhow::anyhow;
+use async_trait::async_trait;
+use serde_json::{json, Value};
+
+use crate::jsonrpc::service::dispatch_infura_request;
+use crate::jsonrpc::types::{HistoryJsonRpcRequest, JsonRequest, Params};
+use crate::portalnet::types::content_key::IdentityContentKey;
+use crate::portalnet::types::messages::ByteList;
+
+// This struct is responsible for dispatching cross-overlay-network requests
+// for data to perform validation. Currently, it just proxies these requests
+// on to infura.
+#[derive(Clone)]
+pub struct HeaderOracle {
```

Comment:
nit: I'd call this `BlockHeaderOracle` to be consistent with other places where `BlockHeader` is used instead of just `Header`, like `ContentKey::BlockHeader`.

Reply:
in conversation during devconnect/syncs we've been throwing around this `HeaderOracle` term, which is a main reason I chose that name. I also considered `ValidationOracle`. Imo the "block" part of "BlockHeader" is somewhat superfluous since there is only one kind of header, therefore no need to distinguish between header types. Also, `ContentKey::BlockHeader` is a defined type from the specs, while this oracle object isn't strictly defined, a bit more general in purpose, and could end up performing various non-header related functions down the road. These are just some thoughts I had for why I went with `HeaderOracle`, but definitely lmk if you still feel strongly in favor of `BlockHeaderOracle`